vendor/github.com/aws/karpenter-core/pkg/controllers/machine/garbagecollection/controller.go

```diff
@@ -261,6 +267,58 @@ func (c *Controller) reconcileMachines(ctx context.Context) error {
 	return multierr.Combine(errs...)
 }
 
+// reconcileCloudProviderInstances is used to garbage leaked cloudprovider isntances.
+func (c *Controller) reconcileCloudProviderInstances(ctx context.Context) error {
+	machineList1 := &v1alpha5.MachineList{}
+	if err := c.kubeClient.List(ctx, machineList1, client.HasLabels([]string{"kaito.sh/workspace"})); err != nil {
+		return err
+	}
+
+	machineList2 := &v1alpha5.MachineList{}
+	if err := c.kubeClient.List(ctx, machineList2, client.HasLabels([]string{"kaito.sh/ragengine"})); err != nil {
+		return err
+	}
+
+	machineListCombined := append(machineList1.Items, machineList2.Items...)
+
+	currentMachineNames := sets.New[string](lo.FilterMap(machineListCombined, func(m v1alpha5.Machine, _ int) (string, bool) {
+		return m.Name, true
+	})...)
+
+	// list all agentpools for garbaging leaked ones.
+	cloudProviderMachines, err := c.cloudProvider.List(ctx)
```

Comment:
Rename cloudProviderMachines to cloudProviderInstances?

Reply:
fixed